tests/foreman/api/test_docker.py

```diff
@@ -688,28 +593,27 @@ def test_publish_once_docker_repo_content_view(self):
         @Feature: Docker
 
         """
-        prod_id = entities.Product(
-            organization=self.org_id
-        ).create_json()['id']
-
-        repo_id = _create_repository(prod_id)['id']
+        product = entities.Product(organization=self.org).create()
+        repo = _create_repository(product)
 
         content_view = entities.ContentView(
-            organization=self.org_id, composite=False).create_json()
-        _add_repo_to_content_view(repo_id, content_view['id'])
+            composite=False,
+            organization=self.org,
+        ).create()
+        content_view.repository = [repo]
+        content_view = content_view.update(['repository'])
+        self.assertIn(repo.id, [repo_.id for repo_ in content_view.repository])
```

Comment:
If reading the content view again is not necessary then I think we should update other places (will check for them after I finish reviewing and will update here)